### ISSUE TITLE
uniform titlebar with toolbar in kvantum

### ISSUE THREAD

**Issue by mojtaba82** [Zren/material-decoration#17]
Hi.
Excuse me for my bad English
I want to opacity and blurness of titlebar and kvantum toolbar be same.
what do I must to do?
when i dont use transparency every  thing is good but when use transparency and use same value of transparency in kvantum and material titlebar, but titlebar and toolbat not become uniform.

**Comment by Zren** [comment]
Issue #8 covers titlebar opacity.

**Comment by mojtaba82** [comment]
This is not duplicate of issue 8.
When I set opacity of material titlebar to a number for example 0.85 and set opacity of kvantum toolbar to same number, titlebar and toolbar not become uniform while when opacity is 1 for both of them, they become uniform.

**Comment by Zren** [comment]
* Which Kvantum theme? What color is the window background? `/usr/share/Kvantum/Materia/`
* How did you set the kvantum opacity?
* `~/.config/kdeglobals` activeBackground=61,62,67

How do you know which color Kvantum is drawning? How do you know which opacity the color is drawn?

**Comment by Zren** [comment]
Have you tried configuring material-decoration's Shadow Size to None? Can you upload a screenshot (use spectacle) of with opacity at 0.8 and opacity at 0.7?

**Comment by mojtaba82** [comment]
> Have you tried configuring material-decoration's Shadow Size to None?

Yes. I have.

> Can you upload a screenshot (use spectacle) of with opacity at 0.8 and another screenshot of opacity at 0.7?

Sure.
opacity at 0.7
![opacity at 0.7](https://user-images.githubusercontent.com/4151722/88960314-92630600-d2b8-11ea-94d2-18770a7135f3.png)

opacity at 0.8
![opacity at 0.8](https://user-images.githubusercontent.com/4151722/88960406-b0306b00-d2b8-11ea-892c-148ce9b14ce5.png)

**Comment by Zren** [comment]
What opacity is KVantum in those screenshots?

**Comment by Zren** [comment]
Also have you tried toggling the background contrast desktop effect?

**Comment by mojtaba82** [comment]
> What opacity is KVantum in those screenshots?

Opened window in above screenshot not have toolbar.
I upload screenshot from a window that have toolbar.

kvantum fluent theme with toolbar color black and 70 alpha. You can see screenshot of svg of fluent in inkscape.
![image](https://user-images.githubusercontent.com/4151722/88962829-3bf7c680-d2bc-11ea-8565-ef1444610b9d.png)

I set opacity of material decoraion to 0.7. but toolbar and titlebar not became uniform.
You can see in below screenshot
![image](https://user-images.githubusercontent.com/4151722/88963318-ecfe6100-d2bc-11ea-897f-21185962a7d8.png)

> Also have you tried toggling the background contrast desktop effect?

Toggling background contrast didn't help.
 I honestly appreciate  you for your attention.

**Comment by Zren** [comment]
Oh the toolbar. Isn't he toolbar svg drawn on top of the window's background?

If the material-decoration is 0.8 opacity, then it would be `255 * 0.8 = 204 = 0xCC` or `#000000cc`. I see you have `255 * 0.7 ~= 178 = 0xB2`. So if you want it to match material's 0.7 opacity, then you need the window background svg + toolbar svg to equal `B2`.

What's the windowOpacity color? `#000000??`

```
windowOpacity = ?
toolbarOpacity = ?
materialOpacity = 178
toolbarOpacity = materialOpacity - windowOpacity
```

**Comment by Zren** [comment]
Ah noticed it's `Fluent-Dark.svg`, so I download it from here:  
https://github.com/Luwx/Fluent-kvantum/blob/master/Fluent-Dark/Fluent-Dark.svg

window-normal is `#212121aa`, uhg it's not pure black. Complicates things. For now I'll focus on the opacity though.

Hexidecimal `aa` is `170` in decimal. So if materialOpacity is `178`, the toolbar opacity needs to be `178 - 170 = 8` or `#00000008`.

... Now that I think of it, why am I trying to fix the toolbar svg. We might as well fix the material decoration opacity + color.

We need material-decoration to draw whatever color `#000000b2` + `#212121aa` is.

**Comment by mojtaba82** [comment]
I wrongly thought that color of toolbar in kvantum is independent from window color.
I found out that it is impossible to find color and opacity that toolbar and titlebar become uniform because color of toolbar is depend on color of window and since window is semi transparent, color of window is depend on color of behind window and color of behind window is out of our control. I did math computing and find out that it is impossible.
only way to uniform toolbar and titlebar is that we use same color in both toolbar and titlebar and dont use transparency in them.
It will become possible  that toolbar and titlebar be semi transparent and uniform only if kvantum draw toolbar independ of window.
I very very thank you for your attention.
